### PR TITLE
Add execution window to PactGroup (fix #31)

### LINKED ISSUE #31: Allow polling up to N pacts concurrently in PactGroup
Similar to [this feature](https://docs.rs/futures/*/futures/stream/trait.StreamExt.html#method.buffered). This is useful when your pacts initiate a heavy operation which you want to limit.

### PR DESCRIPTION
**Pull request by r-darwish**

### CODE REVIEW

**Comment by ayalash** [comment]
I reviewed - No comments, looks great.
@vmalloc Please review & merge.

**Comment by vmalloc** [comment]
This needs documentation. I also don't like the term `execution window` -- perhaps `poll_window`?

File: tests/test_pact_group.py [hunk 1]
```diff
@@ -74,6 +79,29 @@ def get_timeout_exception(self, exc_info):
         group.wait()
 
 
+def test_group_with_poll_window(pred1, pred2, checkpoint1, checkpoint2):
```
Comment (vmalloc) [inline tests/test_pact_group.py]: @r-darwish missing a test checking the assertion for `lazy`+`poll_window`

**Comment by ayalash** [comment]
@vmalloc / @r-darwish - What is the status of this PR?

**Pull request closed without merging**